### ISSUE TITLE
Schema validation accepts empty extensions array but audit rejects it

### ISSUE THREAD

**Issue by chrismlittle123** [chrismlittle123/check-my-toolkit#182]
## Description

`cm validate config` accepts a naming rule with an empty extensions array, but `cm audit` rejects the same configuration. This inconsistency is confusing for users.

## Steps to Reproduce

1. Create a `check.toml` with:
```toml
[code.naming]
enabled = true

[[code.naming.rules]]
extensions = []
file_case = "kebab-case"
folder_case = "kebab-case"
```

2. Run `cm validate config` - passes

3. Run `cm audit` - fails with "Naming rule must have at least one extension"

## Expected Behavior

Both commands should have consistent behavior - either both accept or both reject empty extensions.

Recommendation: Schema validation should reject empty extensions array since it's a meaningless configuration.

## Actual Behavior

- `cm validate config`: `✓ Valid`
- `cm audit`:
```json
{
  "message": "Naming rule must have at least one extension",
  "severity": "error"
}
```

## Impact

Users may create configurations that pass validation but fail audit, leading to confusion.

## Suggested Fix

Add a minimum length check to the extensions array schema validation:
```typescript
extensions: z.array(z.string()).min(1, "At least one extension is required")
```

## Version

check-my-toolkit v1.6.0
